src/components/comment/comment.test.ts

```diff
@@ -0,0 +1,72 @@
+import path from 'path';
+import type { PleasantestUtils } from 'pleasantest';
+import { withBrowser } from 'pleasantest';
+import { loadTwigTemplate } from '../../../test-utils';
+
+const commentMarkup = loadTwigTemplate(path.join(__dirname, 'comment.twig'));
+const initCommentsJs = (utils: PleasantestUtils) =>
+  utils.runJS(`
+    import { initComments } from './comment';
+    initComments();
+  `);
+
+test(
+  'reply form can be opened and closed',
+  withBrowser(async ({ utils, screen, user }) => {
+    await utils.injectHTML(
+      await commentMarkup({
+        comment: {
+          ID: 1,
+          link: '#comment-1',
+          date: new Date(),
+          avatar: '',
+          author: {
+            name: 'Test',
+          },
+          comment_content: 'Test',
+          approved: true,
+        },
+        allow_replies: true,
+      })
+    );
+
+    await utils.loadCSS('../../../dist/standalone.css');
+
+    await initCommentsJs(utils);
+
+    const comment = await screen.getByRole('article');
+    const form = await screen.getByRole('form', { hidden: true });
+    const replyButton = await screen.getByRole('button', {
+      name: /reply/i,
+    });
+
+    // Initial state: reply form is hidden
+    await expect(form).not.toBeVisible();
+    await expect(comment).not.toHaveClass('is-replying');
```

Comment:
In general, I try to avoid using `.toHaveClass`. When I think I need it, I try to ask myself, "how would someone using a screen reader tell if it was in this state", and then I try to use matchers based on the answer to that, rather than `.toHaveClass`.

In this case, I was toggling `.is-replying` in the UI to try to figure out what it does, but I couldn't figure it out. Is it a part of the "public API" of the component (i.e. consumers are supposed to hook into this class)?

Reply:
Good point. `.is-replying` adds the "comment thread" styles (the gray border along the left-hand side which is shown if a comment has replies, or the reply form is open)

From a user perspective I would expect to see a "border" (which is actually a pseudo-element).

<img width="1667" alt="Screen Shot 2021-07-20 at 4 39 51 PM" src="https://user-images.githubusercontent.com/5798536/126408564-b6a2eb4c-2634-43fb-81a8-df3d5e1d3995.png">

I'm not sure of another way to test that 🤔.

I guess I could also have `is-replying` control the visibility of the comment reply form instead of toggling `hidden`. This would simplify my JS and make this easier to test! I'll make that change.

Reply:
Good idea! Is there any accessibility difference between `[hidden]` and `display: none`?

Reply:
The main difference is [`hidden` will work if stylesheets aren't loaded](https://cloudfour.com/thinks/see-no-evil-hidden-content-and-accessibility/#hiding-content-for-everyone). I prefer using `hidden` in general, but I think in this case it would be fine to switch to `display: none;`